torch/_subclasses/functional_tensor.py

```diff
@@ -29,8 +29,36 @@ class FunctionalTensor(torch.Tensor):
     # this is an "infra" mode with lower dispatching precedence.
     _mode_key = torch._C._TorchDispatchModeKey.FUNCTIONAL
 
+    # Note: FunctionalTensor is not quite like the functorch wrapper subclasses,
+    # because today functionalization runs *below* the conj/neg/zero dispatch keys.
+    # We could consider changing this to make it more inline with functorch.
+    _extra_dispatch_keys = torch._C._additional_keys_to_prop_for_wrapper_tensors.add(
+        torch._C.DispatchKey.ZeroTensor
+    )
+
     def __new__(cls, elem):
         assert torch._is_functional_tensor(elem)
+
+        # In general, we'd like our functional tensor subclass to only be in charge of functionalization,
+        # and defer to the inner subclass for all other functionality.
+        # Example: If our inner tensor is a ZeroTensor, we would want to defer running the ZeroTensor fallback
+        # until after we redispatch to our inner ZeroTensor.
+        # However, there are a few keys that we need to mirror between the inner and outer tensors.
+        #   Conjugate
+        #   Negative
+        # Why? These keys are used to test metadata queries, like `.is_conj()` and `.is_neg()`.
+        # We **need** calls to is_conj() to return the same thing on the outer and inner tensors,
+        # Because user code / framework code that branches like so needs to do the same thing
+        # when it sees the outer FunctionalTensor:
+        #     if (x.is_conj()) {
+        #         return at::view_as_real(x.resolve_conj());
+        #     } else {
+        #         return at::view_as_real(x);
+        #     }
+        extra_dispatch_keys = (
+            FunctionalTensor._extra_dispatch_keys & torch._C._dispatch_keys(elem)
+        )
```

Comment:
You're also adding ZeroTensor here. I'm not sure that's what we want (we don't do this in functorch). we probably don't want to add ZeroTensor here, but we do want to re-enable (Conj, Neg, ZeroTensor, Functionalize) later on

Reply:
I actually wanted this, to mirror the fact that the C++ `FunctionalTensorWrapper` will also propagate the ZeroTensor key from the inner tensor to the wrapper ([code](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/FunctionalTensorWrapper.cpp#L36)).

Something does seem fishy about doing this, although it's enough to get tests passing / get parity with C++ functionalization.

Reply:
Fair enough